pkg/apis/workloads/v1alpha1/console_types.go

```diff
@@ -26,44 +26,47 @@ type ConsoleSpec struct {
 	ConsoleTemplateRef corev1.LocalObjectReference `json:"consoleTemplateRef"`
 }
 
+// ConsoleStatus defines the status of a created console, populated at runtime
 type ConsoleStatus struct {
-	PodName    string       `json:"podName" protobuf:"bytes,1,opt,name=podName"`
-	ExpiryTime *metav1.Time `json:"expiryTime,omitempty" protobuf:"bytes,2,opt,name=expiryTime"`
-	Phase      string       `json:"phase" protobuf:"bytes,3,opt,name=phase"`
+	PodName    string       `json:"podName"`
+	ExpiryTime *metav1.Time `json:"expiryTime,omitempty"`
+	Phase      string       `json:"phase"`
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 // ConsoleList is a list of consoles
 type ConsoleList struct {
 	metav1.TypeMeta `json:",inline"`
-	metav1.ListMeta `json:"metadata"`
+	metav1.ListMeta `json:"metadata,omitempty"`
 
 	Items []Console `json:"items"`
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// ConsoleTemplate declares a console template that can be instantiated through a Console object
 type ConsoleTemplate struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	Spec ConsoleTemplateSpec `json:"spec"`
 }
 
+// ConsoleTemplateSpec defines the parameters that a created console will adhere to
 type ConsoleTemplateSpec struct {
-	User                     string                 `json:"user"`
 	Template                 corev1.PodTemplateSpec `json:"template"`
 	DefaultTimeoutSeconds    int                    `json:"defaultTimeoutSeconds"`
 	MaxTimeoutSeconds        int                    `json:"maxTimeoutSeconds"`
-	AdditionalAttachSubjects []rbacv1.Subject       `json:"additionalAttachSubjects"`
+	AdditionalAttachSubjects []rbacv1.Subject       `json:"additionalAttachSubjects,omitempty"`
```

Comment:
You thinking we let you omit this field entirely, and just default it to an empty list?

Reply:
Yeah, I think that this being optional is a nicer interface, and closer to how the built-in resources do it. Let's give this a go - I'm opening to changing it if it becomes a pain down the road.